modules/xmpp/JingleSessionPC.js

```diff
@@ -803,31 +1010,22 @@ export default class JingleSessionPC extends JingleSession {
 
         this.readSsrcInfo(elem);
 
-        const workFunction = finishedCallback => {
+        const workFunction = () => {
             const sdp = new SDP(this.peerconnection.remoteDescription.sdp);
-            const mySdp = new SDP(this.peerconnection.localDescription.sdp);
             const addSsrcInfo = this._parseSsrcInfoFromSourceAdd(elem, sdp);
-
             const newRemoteSdp = this._processRemoteAddSource(addSsrcInfo);
 
-            this._renegotiate(newRemoteSdp)
-                .then(() => {
-                    logger.info('Remote source-add processed');
-                    const newSdp
-                        = new SDP(this.peerconnection.localDescription.sdp);
-
-                    logger.log('SDPs', mySdp, newSdp);
-                    this.notifyMySSRCUpdate(mySdp, newSdp);
-                    finishedCallback();
-                }, error => {
-                    logger.error(
-                        'Error renegotiating after processing remote'
-                            + ` source-add: ${error}`);
-                    finishedCallback(error);
-                });
+            return newRemoteSdp;
         };
 
-        this.modificationQueue.push(workFunction);
+        // Queue and execute
+        this._doRenegotiate('source-add', workFunction)
```

Comment:
`_doRenegotiate` pushes a task to the modification queue and it also contains common/duplicated logic.